features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioPalSecurityManager.cpp

```diff
@@ -607,14 +643,36 @@ bool CordioSecurityManager::sm_handler(const wsfMsgHdr_t* msg) {
             return true;
         }
 
-        case DM_SEC_CALC_OOB_IND:
+        case DM_SEC_CALC_OOB_IND: {
+        	dmSecOobCalcIndEvt_t* evt = (dmSecOobCalcIndEvt_t*) msg;
+        	self._own_oob_present = true;
+        	memcpy(self._confirm, evt->confirm, sizeof(_confirm));
+        	memcpy(self._random, evt->random, sizeof(_random));
+        	handler->on_secure_connections_oob_generated(
+        		address_t(), /* which address should be used ??? */
```

Comment:
Which address should be used ? 
In my opinion, address used should be defined at a much higher level.

Reply:
it is the address with which we advertised, this information is part of the connection event

Reply:
I keep thinking it would be better to not tie oob generation with an address.